src/matrixroomhandler.ts

```diff
@@ -386,6 +386,139 @@ export class MatrixRoomHandler {
     return Promise.reject({err: "Unsupported", code: HTTP_UNSUPPORTED});
   }
 
+  public async HandleDiscordCommand(msg: Discord.Message) {
+    if (!msg.member.hasPermission("ADMINISTRATOR")) {
+      msg.channel.sendMessage("**ERROR:** insufficiant permissions to use matrix commands");
+      return;
+    }
+    if (!(<Discord.TextChannel> msg.channel).guild) {
+      msg.channel.sendMessage("**ERROR:** only available for guild channels");
+    }
+    const prefix = "!matrix ";
+    let command = "help";
+    let args = [];
+    if (msg.content.length >= prefix.length) {
+      const allArgs = msg.content.substring(prefix.length).split(" ");
+      if (allArgs.length && allArgs[0] !== "") {
+        command = allArgs[0];
+        allArgs.splice(0, 1);
+        args = allArgs;
+      }
+    }
+    
+    let replyMessage = "Error, unkown command. Try `!matrix help` to see all commands";
+    
+    const intent = this.bridge.getIntent();
+    const doAction = async (funcKey, action) => {
+      const name = args.join(" ");
+      const channels = await this.discord.GetRoomIdsFromChannel(msg.channel);
+      try {
+        const userMxid = await this.GetMxidFromName(name, channels);
+        let allChannels = [];
+        /* tslint:disable:await-promise */
```

Comment:
[You can certainly lint rules for one line](https://github.com/Half-Shot/matrix-appservice-discord/blob/master/src/discordas.ts#L24), btw.